pydantic_core/core_schema.py

```diff
@@ -1447,16 +1492,77 @@ def __call__(self, __input_value: Any, __info: ValidationInfo) -> Any:  # pragma
         ...
 
 
+class ModelFieldValidatorFunction(Protocol):
+    def __call__(self, __input_value: Any, __info: ModelFieldValidationInfo) -> Any:  # pragma: no cover
+        ...
+
+
+class ModelFieldValidatorFunctionSchema(TypedDict):
+    call: Required[ModelFieldValidatorFunction]
```

Comment:
as per my other comment, I marginally in favour of switching to a tuple, but if we stick with a typed-dict, should be 
```suggestion
    call: ModelFieldValidatorFunction
```
and same below.

Reply:
I think keeping the typed dict is best. I moved `type` to first field and removed the Required.